src/storage/src/hummock/sstable/sstable_id_manager.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::VecDeque;
+use std::ops::DerefMut;
+use std::sync::Arc;
+
+use risingwave_hummock_sdk::HummockSstableId;
+use risingwave_rpc_client::HummockMetaClient;
+use tokio::sync::Mutex;
+
+use crate::hummock::{HummockError, HummockResult};
+
+pub type SstableIdManagerRef = Arc<SstableIdManager>;
+
+/// 1. Caches SST ids fetched from meta.
+/// 2. TODO #4037: Maintains watermark SST ids. It will be used by SST full GC.
+pub struct SstableIdManager {
+    unused_sst_ids: Mutex<VecDeque<HummockSstableId>>,
+    remote_fetch_number: u32,
+    hummock_meta_client: Arc<dyn HummockMetaClient>,
+}
+
+impl SstableIdManager {
```

Comment:
1. can we use `begin_table_id` and `segment` to replace VecDeque it looks more easier to implement
2. after 1, we can simply do some async prefetch to avoid running out of local table_id (if needed)

Reply:
Fixed 1.
I'd like to defer 2 to keep code simple until necessary.